### PR TITLE
Allow document from other server, add allow list for document rule

### LINKED ISSUE #1155: Checking compound document is not compatible with Echidna
The [linkchecker rule](https://github.com/w3c/specberus/blob/main/lib/rules/links/linkchecker.js) checks if all the resources are either in the same directory or in the allow list.
This is a problem for Echidna as it installs the document in a temporary web server to run the checks. Hence the document is not at its final location when Echidna runs pubrules.

We don't want to hardcode the echidna web server URLs in the allow list so we need a way to pass a config option so pubrules will know that the resources URLs will not start by the 'This Version' URL.

### PR DESCRIPTION
**Pull request by jennyliang220**
Closes #1168 , resolve #1155

### CODE REVIEW

File: lib/rules/links/linkchecker.js [hunk 1]
```diff
@@ -14,8 +14,8 @@ const compound = {
 };
 
 const allowList = [
-    new RegExp('^https://www.w3.org/StyleSheets/TR/2016/'),
-    new RegExp('^https://www.w3.org/scripts/'),
+    new RegExp('^https://www.w3.org/StyleSheets/TR/'),
+    new RegExp('^https://www.w3.org/scripts/TR/'),
```
Comment (deniak) [inline lib/rules/links/linkchecker.js:18]: We can allow any scripts under `/scripts/`. We do have control over that folder.
```suggestion
    new RegExp('^https://www.w3.org/scripts/'),
```

File: lib/rules.json [hunk 2]
```diff
@@ -1868,7 +1868,7 @@
             "compound": {
                 "name": "8. Compound Documents",
                 "rules": {
-                    "compoundFilesLocation": "If the document is compound (i.e., if it consists of more than one file), all the files <span class=\"rfc2119\">must</span> be under a directory /TR/YYYY/@{param1}-shortname-YYYYMMDD/",
+                    "compoundFilesLocation": "If the document is compound (i.e., if it consists of more than one file), all the files <span class=\"rfc2119\">must</span> be under a directory /TR/YYYY/@{param1}-shortname-YYYYMMDD/. Exceptions are resources under these paths:<ol><li>https://www.w3.org/StyleSheets/TR/</li><li>https://www.w3.org/scripts/TR/</li></ol>",
```
Comment (deniak) [inline lib/rules.json:1871]:
```suggestion
                    "compoundFilesLocation": "If the document is compound (i.e., if it consists of more than one file), all the files <span class=\"rfc2119\">must</span> be under a directory /TR/YYYY/@{param1}-shortname-YYYYMMDD/. Exceptions are resources under these paths:<ol><li>https://www.w3.org/StyleSheets/TR/</li><li>https://www.w3.org/scripts/</li></ol>",
```

File: lib/rules/links/linkchecker.js [hunk 3]
```diff
@@ -47,28 +47,29 @@ exports.check = async function (sr, done) {
...
-    const thisLink = dts && dts.This && dts.This.dd.querySelector('a').href;
-    if (!thisLink) {
+    // sr.url is used as base url. Every other resources should use in same folder as base. e.g.
+    // - spec doc: https://www.w3.org/TR/2021/WD-pubrules-20210401/
+    // - image (pass): https://www.w3.org/TR/2021/WD-pubrules-20210401/images/sample.png
+    // - image (pass): https://www.w3.org/TR/2021/WD-pubrules-20210401/sample.png
+    // - image (error): https://w3c.github.io/pubrules/sample.png
+
+    if (!sr.url) {
+        await browser.close();
```
Comment (deniak) [inline lib/rules/links/linkchecker.js:57]: As you mentioned in our last call, you can move the launch of puppeteer after that `if`

**Review by deniak — CHANGES_REQUESTED** [review]

**Review by deniak — APPROVED** [review]

**Review by Nuckijfuta — COMMENTED** [review]
Revert

**Pull request merged**